Fix build process and add buffered shaders

Doridian opened the pull request:
For now this builds, just keeping it here to keep track of the changes and to make giving feedback easier :3

TODO, mostly for my own sanity:
- [x] Make all shaders compile, main thing missing is support for the `meta.json` with its dynamic `#define`-s
- [ ] Work out process to put the QSB files in the right places
- [ ] Make Panon load QSBs and not the old style
- [x] Rip out QSB from Panon entirely
- [x] Make parameters work
- [ ] Make parameters less ugly (?)

Doridian committed:
Fix build process and add buffered shaders

Doridian committed:
rm

Doridian committed:
Move shaders, some compile now

flafflar commented:
Hey @Doridian, could you add me as a collaborator in your fork? I want to help with the development. Normally I would give you access to my repository, but since you have already created the pull request I think it's simpler this way. We can use the pull request to coordinate. Do as you please :3

Doridian commented:
@flafflar You should have write access to the `qsb` branch in my repo (due to "Allow edits by maintainers"), I think?
But I've also invited you as a collaborator on my fork.

flafflar commented:
> You should have write access to the qsb branch in my repo (due to \"Allow edits by maintainers\"), I think?

Oh, I didn't know that that was a thing o~o

flafflar committed:
Move Python scripts inside plasmoid directory

CMake copied the symlink instead of the directory when installing,
which made it invalid.

flafflar committed:
Load the QSBs instead of the shaders

Doridian committed:
Make GLDFT and wave-buffer compile. Those shaders require OpenGL 1.5, but aren't currently in-use anyway

Doridian committed:
Remove a bunch of unused imports

Doridian committed:
Sort of ugly hack, but this should make parameter inputs to shaders work once again

Doridian committed:
Make Spectrum.qml actually load

Doridian committed:
qsb in right install for now

Doridian committed:
And now effects load

Doridian committed:
Rip shader source out of plasmoid

Doridian committed:
Provide subfields

Doridian committed:
Configs now actually load

Doridian commented:
Recently I got some more drive to actually push some more stuff.
This current build can now successfully load shaders once again.
However something is horribly messed up with the way settings get sent from panon to the shader (the settings UI itself works just fine), making bar1ch completely invisible and unable to alter the behavior of spectrogram. I'll check that out next.

Doridian committed:
And boom, fixed

Doridian commented:
And config loading is fixed, too!

Doridian committed:
fixes

Doridian committed:
Revert "fixes"

This reverts commit f1af026cb88cbda4348ecf8461f91d6b55f9cb18.

Doridian committed:
fix

Doridian committed:
One CMakeList to rule them all

Doridian committed:
dont install

Doridian committed:
pkg the end

Doridian committed:
More easier build system

Doridian committed:
Allow cleanbuild as option

Doridian committed:
Install should cleanbuild

Doridian committed:
Makepackage is a misleading name

Doridian committed:
But add it back as a symlink to make it clear

Doridian committed:
Dont gitignore these so its obvious if theres extra files

Doridian committed:
Ah well,this works

Doridian committed:
These dont make sense

Doridian commented:
Alright, one and a half bug left:
- The `chain` effect doesn't work at all.
- `bar1ch` peaks act a bit weirdly but seems functional

Doridian committed:
Fix build

Doridian committed:
These are all readonly
